docs/data/data-grid/prompt/PromptWithExamples.tsx

```diff
@@ -0,0 +1,98 @@
+import * as React from 'react';
+import {
+  DataGridPremium,
+  Unstable_GridToolbarPromptControl as GridToolbarPromptControl,
+  GridToolbar,
+} from '@mui/x-data-grid-premium';
+import {
+  mockPromptResolver,
+  randomBoolean,
+  randomCompanyName,
+  randomCountry,
+  randomCreatedDate,
+  randomEmail,
+  randomInt,
+  randomJobTitle,
+  randomPhoneNumber,
+  randomTraderName,
+  useDemoData,
+} from '@mui/x-data-grid-generator';
+import Stack from '@mui/material/Stack';
+import Box from '@mui/material/Box';
+
+function ToolbarWithPromptInput() {
```

Comment:
Since the prompt control will usually be in its own part of the layout (sometimes even completely apart from our standard buttons) I would prefer to keep it as a separate component

It is also aligned with [our direction to make the toolbar more composable](https://github.com/mui/mui-x/pull/14611)